code/game/turfs/open/_open.dm

```diff
@@ -107,6 +113,52 @@
 	. = ..()
 	update_visuals()
 
+//ATTACK GHOST IGNORING PARENT RETURN VALUE
+/turf/open/attack_ghost(mob/dead/observer/user)
+	if(destination_z)
+		var/turf/T = locate(destination_x, destination_y, destination_z)
+		user.forceMove(T)
+
+/turf/open/proc/CanBuildHere()
+	if(destination_z)
+		return FALSE
+	return TRUE
+
+/turf/open/space/is_transition_turf()
+	if(destination_x || destination_y || destination_z)
+		return TRUE
+
+/turf/open/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	. = ..()
+	if(!arrived || src != arrived.loc)
+		return
+
+	if(destination_z && destination_x && destination_y && !arrived.pulledby && !arrived.currently_z_moving)
+		var/tx = destination_x
+		var/ty = destination_y
+		var/turf/DT = locate(tx, ty, destination_z)
+		var/itercount = 0
+		while(DT.density || istype(DT.loc,/area/shuttle)) // Extend towards the center of the map, trying to look for a better place to arrive
+			if (itercount++ >= 100)
+				log_game("SPACE Z-TRANSIT ERROR: Could not find a safe place to land [arrived] within 100 iterations.")
+				break
+			if (tx < 128)
+				tx++
+			else
+				tx--
+			if (ty < 128)
+				ty++
+			else
+				ty--
+			DT = locate(tx, ty, destination_z)
```

Comment:
seems like a poor solution for this particular usecase

Reply:
yea I know and I don't have a GREAT solution outside of transitions to dense turfs blocking the move, buttt